src/newViews/ProjectEdit/GeoAreas/RegionCard/AddAdminLevel/AddAdminLevelForm/index.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { useCallback, useMemo } from 'react';
+import { isDefined } from '@togglecorp/fujs';
+
+import {
+    MdFileUpload,
+} from 'react-icons/md';
+import {
+    Button,
+    NumberInput,
+    TextInput,
+    SelectInput,
+    FileInput,
+    ContainerCard,
+} from '@the-deep/deep-ui';
+import {
+    ObjectSchema,
+    PartialForm,
+    requiredStringCondition,
+    useForm,
+    getErrorObject,
+} from '@togglecorp/toggle-form';
+
+import notify from '#notify';
+import { useRequest, useLazyRequest } from '#utils/request';
+import {
+    AdminLevelGeoArea,
+    MultiResponse,
+} from '#typings';
+
+import styles from './styles.scss';
+
+type FormType = {
+    title: string;
+    level?: number;
+    nameProp?: string;
+    codeProp?: string;
+    region: number;
+    parent?: number;
+    geoShapeFile: number;
+};
+
+type FormSchema = ObjectSchema<PartialForm<FormType>>;
+type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+const schema: FormSchema = {
+    fields: (): FormSchemaFields => ({
+        title: [requiredStringCondition],
+        level: [],
+        nameProp: [],
+        codeProp: [],
+        region: [],
+        parent: [],
+        geoShapeFile: [],
+    }),
+};
+
+const adminLevelKeySelector = (d: AdminLevelGeoArea) => d.id;
+const adminLevelLabelSelector = (d: AdminLevelGeoArea) => d.title;
+
+interface Props {
+    activeRegion: number;
+    onSuccess: () => void;
+}
+
+function AddAdminLevelForm(props: Props) {
+    const {
+        activeRegion,
+        onSuccess,
+    } = props;
+
+    const formValue: PartialForm<FormType> = useMemo(() => ({
+        region: activeRegion,
+    }), [activeRegion]);
+
+    const {
+        pristine,
+        value,
+        error: riskyError,
+        setFieldValue,
+        validate,
+        setError,
+    } = useForm(schema, formValue);
+
+    const error = getErrorObject(riskyError);
+
+    const adminLevelQuery = useMemo(() => ({
+        region: activeRegion,
+    }), [activeRegion]);
+
+    const {
+        pending: addAdminLevelPending,
+        trigger: addAdminLevelTrigger,
+    } = useLazyRequest({
+        url: 'server://admin-levels/',
+        method: 'POST',
+        query: adminLevelQuery,
+        body: ctx => ctx,
+        onSuccess: () => {
+            onSuccess();
+            notify.send({
+                title: 'Post Admin Levels',
+                type: notify.type.SUCCESS,
+                message: 'Successfully posted admin levels',
```

Comment:
We can improve the success message.

Reply:
Do you mean using the translations or any other way?